python/tank/util/metrics.py

```diff
@@ -345,79 +348,164 @@ def data(self):
         return self._data
 
 
-class UserActivityMetric(ToolkitMetric):
-    """Convenience class for a user activity metric."""
+class EventMetric(ToolkitMetric):
+    """
+    Convenience class for creating a metric event.
+
+    Use this helper class to create a suitable metric structure that you can
+    then pass to the `tank.utils.metrics.log_metric_event` method.
 
-    def __init__(self, module, action):
-        """Initialize the metric with the module and action information.
-        
-        :param str module: Name of the module in which action was performed.
-        :param str action: The action that was performed.
-        
+    The simplest usage of this class is simply to provide an event group and
+    event name to the constructor:
+
+    Optionally, you can add your own specific metrics by using the
+    `add_event_property` method. The later simply takes a key-value
+    pair and properly structure the entry into the metric structure.
+
+    metric.add_event_property("MyAppVersion", "10.13.1")
+    metric.add_event_property("ViewedPageCount", 99 )
+
+    Optionally, you can add system information by calling the
+    `add_system_info_properties` method. The method takes care of gathering
+    relevant information and properly formatting it for submission to a
+    Shotgun site.
+
+    Also optionally, you can add user information by calling the
+    `add_user_info_properties`. The method takes care of gathering relevant
+    information regarding location, country, used language and properly
+    formatting it for submission to a Shotgun site.
+
+    Below is a complete typical usage:
+
+    ```
+    metric = EventMetric(event_group="App", event_name="Logged In")
+    metric.add_event_property("MyAppVersion", "10.13.1")
+    metric.add_event_property("ViewedPageCount", 99 )
+    metric.add_system_info_properties()
+    metric.add_user_info_properties()
+    log_event_metric(metric)
+    ```
+
+    TODO: How about event type which is defined in the 'Shotgun Event Taxonomy' document as event property???
```

Comment:
I'm referencing the Ruby server code which mention a "event_type" inside of the "event_property" structure: https://github.com/shotgunsoftware/shotgun/blob/master/app/controllers/api3_controller.rb#L331